cdk/l3-cdk-resources/atlas-private-endpoint/src/index.ts

```diff
@@ -0,0 +1,139 @@
+import {ApiKeyDefinition, AtlasBasic, AtlasBasicProps, ProjectProps} from "@mongodbatlas-awscdk/atlas-basic";
+import {Construct} from "constructs";
+import {
+    ApiKey,
+    CfnPrivateEndpoint,
+    CfnPrivateEndpointProps, PrivateEndpoint,
+} from "@mongodbatlas-awscdk/private-endpoint";
+
+/** @type {*} */
+const privateEndpointDefaults = {
+    region: 'us-east-1'
+};
+
+/**
+ * @description
+ * @export
+ * @class AtlasBasicPrivateEndpoint
+ * @extends {Construct}
+ */
+export class AtlasBasicPrivateEndpoint extends Construct {
+
+    readonly atlas : AtlasBasic;
+    readonly private_endpoint : CfnPrivateEndpoint;
+
+    /**
+     * Creates an instance of AtlasBasicPrivateEndpoint.
+     * @param {Construct} scope
+     * @param {string} id
+     * @param {AtlasPrivateEndpointProps} props
+     * @memberof AtlasBasicPrivateEndpoint
+     */
+    constructor(scope: Construct, id: string, props: AtlasPrivateEndpointProps) {
```

Comment:
can we add validation of the input so that the constructor will throw an exception if a required field is undefined or empty string? we want to fail fast so that the user does not have to wait for the stack to run to get an error

Reply:
added the validation. for AtlasBasic it would be updated in the next release.